src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java

```diff
@@ -171,47 +168,52 @@ public final HostConnectionCounters getHostConnectionCounters() {
     return accessibleDatacenters;
   }
 
-  public Optional<JmxCredentials> getJmxCredentialsForCluster(String clusterName) {
```

Comment:
can we keep the `getJmxCredentialsForCluster(String clusterName)` method as well? it is the inner block to the `getJmxCredentialsForCluster(Node node)` method, and is used by the `ClusterResource.getCluster(..)` rest endpoint. (It would be nice if that endpoint method didn't have to pass in a blank node as a hack.)

Reply:
It's not entirely a blank node.  It's a node with a cluster but without a hostname.  The cluster is needed because of the potential JmxCredentials that have been set.  I can change this to allow clusterName again but it would need to then also look up the Cluster object.

Reply:
I think it's ok to look up the cluster object. Here it's happening at a high-level, just once in the request, in the `ClusterResource` class.

Reply:
looking closer, the ClusterResource really does not need to pass the Node into getJmxCredentialsForCluster.  All he really needs is the Cluster object.  Knowing the hostname does not effect the credentials.  How about getJmxCredentialsForCluster accepts an @Nullable Cluster instead of a Node?

Reply:
`Optional<Cluster>` please :)

`@Nullable` shouldn't be used for public method signatures, except those json serialisable.